bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -7772,6 +7772,40 @@ public void setImagePaths(long obj_id, List<String> paths) throws TskCoreExcepti
 			releaseSingleUserCaseWriteLock();
 		}
 	}
+    /**
+	 * Deletes a datasource from the open case, the database has foreign keys with a delete cascade
+	 * so that all the tables that have a datasource object id will have their data deleted.
+	 * 
+	 * @param dataSourceObjectid id of Datasource to be Deleted
+	 * 
+	 * @throws TskCoreException exception thrown when critical error occurs
+	 *                          within tsk core and the update fails
+	 */
+	public void deleteDataSource(long dataSourceObjectId) throws TskCoreException {
+        CaseDbConnection connection = connections.getConnection();
+		Statement statement = null;
+		acquireSingleUserCaseWriteLock();
+		try {
+			statement = connection.createStatement();
+			connection.beginTransaction();
+			// The following delete(s) uses a foreign key delete with cascade in the DB so that it will delete
+			// all associated rows from tsk_object and its children.  For large data sources this may take some time.
+			statement.execute("DELETE FROM tsk_objects WHERE obj_id = " + dataSourceObjectId);
+			// The following delete uses a foreign key delete with cascade in the DB so that it will delete all
+			// associated rows from accounts table and its children.
+			String accountSql = "DELETE FROM accounts where account_id in (SELECT account_id FROM accounts " +
+                                "WHERE account_id NOT IN (SELECT account1_id FROM account_relationships) " +
+                                "AND account_id NOT IN (SELECT account2_id FROM account_relationships))";
+            statement.execute(accountSql);
+			connection.commitTransaction();
+		} catch (SQLException ex) {
+				connection.rollbackTransaction();
+			    throw new TskCoreException("Error deleting data source.", ex);	
```

Comment:
It would be better to include the data source object ID in this log message to aid in debugging.

Reply:
Added datasource id to log message and to comments.